rest/data/commit_queue.go

```diff
@@ -236,6 +251,19 @@ func (pc *MockCommitQueueConnector) CommitQueueClearAll() (int, error) {
 	return count, nil
 }
 
+func (pc *MockCommitQueueConnector) IsPatchEmpty(id string) (bool, error) {
+	patchDoc, err := patch.FindOne(patch.ById(patch.NewId(id)).WithFields(patch.PatchesKey))
```

Comment:
instead of querying the db a test can add to the mock connector's CachedPatches field